### PR TITLE
Implement missing status checks

### PR DESCRIPTION
**Pull request by huy-nodeset**

### CODE REVIEW

File: common/nodeset-client-v1.go [hunk 1]
```diff
@@ -554,12 +556,17 @@ func IsUploadedToNodeset(pubKey beacon.ValidatorPubkey, registeredPubkeys []beac
...
-	return false
-}
-
-func IsUploadedStakewise(pubKey beacon.ValidatorPubkey, statuses map[beacon.ValidatorPubkey]beacon.ValidatorStatus) bool {
-	// TODO: Implement
-	return false
+func GetNodesetStatus(pubKey beacon.ValidatorPubkey, registeredPubkeysStatusMapping map[beacon.ValidatorPubkey]string) swtypes.NodesetStatus {
+	for registeredPubKey, nodesetStatus := range registeredPubkeysStatusMapping {
+		if registeredPubKey == pubKey {
+			if nodesetStatus == "REGISTERED" {
```
Comment (jclapis) [inline common/nodeset-client-v1.go:562]: Making a note that we'll have to revisit this later to turn these into enum values once they've been solidified

File: go.mod [hunk 2]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/google/uuid v1.6.0
 	github.com/gorilla/mux v1.8.1
 	github.com/nodeset-org/hyperdrive-daemon v0.5.0-b1.0.20240614184837-e4f7a44efd10
-	github.com/nodeset-org/nodeset-svc-mock v0.1.1-0.20240614184020-4f7aeb08967e
+	github.com/nodeset-org/nodeset-svc-mock v0.1.1-0.20240617160247-3ac8bfac0cb9
```
Comment (jclapis) [inline go.mod:14]: We can update this to v0.2.0 now.

File: internal/tests/api/status_test.go [hunk 3]
```diff
@@ -71,6 +71,11 @@ func TestValidatorStatus_Active(t *testing.T) {
 	require.NoError(t, err)
 	t.Log("Validator added to the beacon chain")
 
+	// Mark the validator as active
+	err = nsMock.MarkValidatorRegistered(vault, network, depositDataSet)
+	require.NoError(t, err)
+	t.Log("Deposit data set marked as uploaded")
```
Comment (jclapis) [inline internal/tests/api/status_test.go:77]: This should say registered instead of uploaded.

**Review by jclapis — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (huy-nodeset) [reply go.mod]: done

File: internal/tests/api/status_test.go (hunk 3 above)
Comment (huy-nodeset) [reply internal/tests/api/status_test.go]: done

File: common/nodeset-client-v1.go (hunk 1 above)
Comment (huy-nodeset) [reply common/nodeset-client-v1.go:563]: done

**Review by jclapis — APPROVED** [review]

**Pull request merged**